src/sentry.ts

```diff
@@ -21,7 +21,52 @@ import { MatrixClientPeg } from "./MatrixClientPeg";
 import SettingsStore from "./settings/SettingsStore";
 import { MatrixClient } from "matrix-js-sdk";
 
-async function getStorageOptions(): Promise<Record<string, string>> {
+/* eslint-disable camelcase */
+
+type StorageContext = {
+    storageManager_persisted?: string;
+    storageManager_quota?: string;
+    storageManager_usage?: string;
+    storageManager_usageDetails?: string;
+};
+
+type UserContext = {
+    username: string;
+    enabled_labs: string;
+    low_bandwidth: string;
+};
+
+type CryptoContext = {
```

Comment:
Just a heads up, you can instead use `Partial<type>` where you had all the types optional and don't need to repeat `?` everywhere.